extension-manifest-v3/src/background/cosmetics.js

```diff
@@ -205,13 +199,13 @@ async function adblockerOnMessage(msg, sender, sendResponse) {
     sendResponse({
       active: specificResponses.map((r) => r.active).some((a) => a),
       extended: specificResponses.map((r) => r.extended).flat(),
-      styles: '',
+      scripts: specificResponses.map((r) => r.scripts).flat(),
```

Comment:
this is removed by purpose as we inject scripts with webNavigation listener https://github.com/ghostery/ghostery-extension/commit/75fd18ad98676528782fb5dc9a9f6eb1038d1bd8#diff-b70e3b1b9a717375c4c1fedc6bb7c6e2fca37465d4c4b2f0d223579ecb4a4beaL203

Reply:
It is not the same - https://github.com/ghostery/ghostery-extension/blob/fix-mv3-cosmetics/extension-manifest-v3/src/background/cosmetics.js#L281

We use other flags in the generation, so I assume the `scripts` list can differ.

Reply:
@remusao  maybe we can keep both? it should not harm, right?

Reply:
I'd support what Philipp say, we should inject scriptlets only once per frame. If it's already done with webNavigation listener then let's not do it again from content script.

Reply:
So... @chrmod @remusao @philipp-classen Can you answer a simple question - is there a difference between those two calls to the API:

```
const { scripts } = engine.getCosmeticsFilters({ ..., getExtendedRules: true, ... });
```

And:
```
const { scripts } = engine.getCosmeticsFilters({ ..., getExtendedRules: false, ... });
```

If it does, then our injected scripts on navigate are not the same as those generated when the content script sends `getCosmeticsFilters` message.

If you answer this question, then we have the following options:
1. Remove condition starting at L174 completely
2. accept the change
3. Make correct change - as maybe those scripts should be passed in another way?

Reply:
Extended rules (`getExtendedRules`) should have nothing to do with scripts so these two calls should be equivalent (in fact I think these are disabled in Ghostery's adblocker so it does not matter which value you give to `getExtendedRules`).

Reply:
Cool, thanks for the answer, so we can remove condition completely :)

Reply:
I'll make a change

Reply:
@remusao @chrmod @philipp-classen I have started cleaning up the condition, and if it is really not required, then in that case, this background process should not respond to the content script. What worries me, is that if there are no `scripts`, then the returned `extended` is not also passed there, but I think it should.

I check the adblocker source code, and those `extended` seems to be styling rules:
https://github.com/ghostery/adblocker/blob/master/packages/adblocker/src/engine/bucket/cosmetic.ts#L565

Currently, they are passed, only if there are also scripts, but those scripts are not included, but only `extended`. Is it anyway correct? Are those `extended` related to `scripts`?

Reply:
https://github.com/ghostery/adblocker-filters/blob/main/src/configs.ts#L121

We have disabled extended selectors for Ghostery so these will always be empty. Also, there is not relation between extended and scripts, they are totally independent features (one can be enabled without the other).